pkg/oplabels/labels.go

```diff
@@ -32,6 +32,18 @@ func AddOperatorLabels(m map[string]string, dc *api.CassandraDatacenter) {
 			m[key] = api.CleanLabelValue(value)
 		}
 	}
+
+}
+
+func AddOperatorAnnotations(m map[string]string, dc *api.CassandraDatacenter) {
+	if m == nil {
+		m = make(map[string]string)
+	}
+	if len(dc.Spec.AdditionalAnnotations) != 0 {
+		for key, value := range dc.Spec.AdditionalAnnotations {
+			m[key] = api.CleanLabelValue(value)
```

Comment:
This isn't correct. Label values have different requirements in Kubernetes than Annotation values (label and annotations have similar key requirements, but not values).

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Reply:
That's interesting, I didn't realise, I'll correct this.

Reply:
It looks like (although it isn't mentioned directly in those links) annotation values are basically a free-for-all. I guess that makes sense as they aren't indexed the same way. I'll remove all validation since it doesn't seem like it will be required.